### PR TITLE
added to base station section

### PR DESCRIPTION
**Pull request by Arthur-H21**
Added the choice of using certain libraries.

### CODE REVIEW

File: docs/design/3_1_6_base_station.tex [hunk 1]
```diff
@@ -18,11 +18,19 @@ \subsubsection{Functionality}
...
+Radio buttons and Sliders are situated on the web-application for user input (\textbf{NF.BS.2}). Their functions include adjusting the visual properties of the web-application and changing settings on the PMB and PLB (\textbf{F.BS.4}) such as adjusting the camera polling rate and ML model sensitivity.
+
+\subsubsection{Programming Libraries}
+
+There are two JavaScript libraries used in both the client-side and server-side code of the base station.
+
+The first library is the socket.io library, which enables real-time, bidirectional and event-based communication between browser and server. This library is used to ensure a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station.
```
Comment (FSXAC) [inline docs/design/3_1_6_base_station.tex:34]:
```suggestion
The first library is \textit{socket.io}, which enables real-time, bidirectional and event-based communication between browser and server. We use socket.io to establish a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station. 
```

File: docs/design/3_1_6_base_station.tex [hunk 2]
```diff
@@ -18,11 +18,19 @@ \subsubsection{Functionality}
...
+
+\subsubsection{Programming Libraries}
+
+There are two JavaScript libraries used in both the client-side and server-side code of the base station.
+
+The first library is the socket.io library, which enables real-time, bidirectional and event-based communication between browser and server. This library is used to ensure a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station. 
+
+The second library is the p5.js library, which is a client-side JavaScript platform that provides programmers with tools to create virtual canvases more easily. This library is used to handle live video feed and ML bounding box data display; it allows for an elegant and high-level approach to programming this part of the web application.
```
Comment (FSXAC) [inline docs/design/3_1_6_base_station.tex:36]:
```suggestion
The second library is \textit{p5.js}, a client-side JavaScript framework that provides programmers with tools to create virtual canvases more easily. We use p5.js to draw live video feed and ML bounding box data; it allows for an elegant and high-level approach to programming this part of the web application. 
```

File: docs/design/3_1_6_base_station.tex [hunk 3]
```diff
@@ -18,11 +18,19 @@ \subsubsection{Functionality}
...
 
-Buttons are situated on the web-application for user input (\textbf{NF.BS.2}). Their functions include adjusting the visual properties of the web-application and changing settings on the PMB and PLB (\textbf{F.BS.4}) such as adjusting the camera polling rate and ML model sensitivity.
+The user may use buttons and sliders on the web-application to adjust the visual properties of the web-application or change the settings on the PMB and PLB (\textbf{F.BS.2, F.BS.4}) such as the camera polling rate and ML sensitivity.
+
+\subsubsection{Programming Libraries}
+
+There are two JavaScript libraries used in both the client-side and server-side code of the base station.
+
+The first library is \textit{socket.io}, which enables real-time, bidirectional and event-based communication between browser and server. We use socket.io to establish a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station.
```
Comment (peterdeutsch) [inline docs/design/3_1_6_base_station.tex:34]:
```suggestion
The first library is \textit{socket.io}, which enables real-time, bidirectional and event-based communication between the browser and server. We use socket.io to establish a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station. 
```

File: docs/design/3_1_6_base_station.tex [hunk 4]
```diff
@@ -18,11 +18,19 @@ \subsubsection{Functionality}
...
+The user may use buttons and sliders on the web-application to adjust the visual properties of the web-application or change the settings on the PMB and PLB (\textbf{F.BS.2, F.BS.4}) such as the camera polling rate and ML sensitivity.
+
+\subsubsection{Programming Libraries}
+
+There are two JavaScript libraries used in both the client-side and server-side code of the base station.
+
+The first library is \textit{socket.io}, which enables real-time, bidirectional and event-based communication between browser and server. We use socket.io to establish a single communication protocol between the base station server and its clients. The client-side portion of the library is available in other languages such as C++ and Python with the same communication protocol, allowing for more versatility when programming the PMB to communicate with the base station. 
+
+The second library is \textit{p5.js}, a client-side JavaScript framework that provides programmers with tools to create virtual canvases more easily. We use p5.js to draw live video feed and ML bounding box data; it allows for an elegant and high-level approach to programming this part of the web application.
```
Comment (peterdeutsch) [inline docs/design/3_1_6_base_station.tex:36]: I didn't even know we were using this, turns out this document might be useful after all ;)

**Pull request merged**